### PR TITLE
Add gradle check test for github workflows (Testing again)

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/workflows/code-hygiene.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Code Hygiene
 
-on: [push, pull_request]
```
Comment (dreamer-89) [inline .github/workflows/code-hygiene.yml:3]: I searched for different events and went through different activity types in [pull_request](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) event but didn't fine one which triggers workflow on commit; while [push](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#push) specifically does that. Am I missing something ?

Comment (peterzhuamazon) [reply .github/workflows/code-hygiene.yml:3]: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
* synchronize

**Comment by dreamer-89** [comment]
Flaky test failure. 
```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.indices.replication.SegmentReplicationTargetServiceTests.testBeforeIndexShardClosed_CancelsOngoingReplications" -Dtests.seed=7063FEA65C9F22BE -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=is-IS -Dtests.timezone=Europe/San_Marino -Druntime.java=17

org.opensearch.indices.replication.SegmentReplicationTargetServiceTests > testBeforeIndexShardClosed_CancelsOngoingReplications FAILED
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=663, name=opensearch[org.opensearch.indices.replication.SegmentReplicationTargetServiceTests][generic][T#1], state=RUNNABLE, group=TGRP-SegmentReplicationTargetServiceTests]
        at __randomizedtesting.SeedInfo.seed([7063FEA65C9F22BE:208DF90D4830F75]:0)

        Caused by:
        java.lang.NullPointerException: Cannot invoke "org.opensearch.indices.replication.common.ReplicationCollection$ReplicationRef.get()" because "replicationRef" is null
            at __randomizedtesting.SeedInfo.seed([7063FEA65C9F22BE]:0)
            at org.opensearch.indices.replication.SegmentReplicationTargetService.start(SegmentReplicationTargetService.java:175)
            at org.opensearch.indices.replication.SegmentReplicationTargetService$ReplicationRunner.run(SegmentReplicationTargetService.java:169)
            at org.opensearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:739)
            at java.****/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
            at java.****/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
            at java.****/java.lang.Thread.run(Thread.java:833)

```

File: .github/workflows/gradle-check.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+name: Jenkins Gradle Check
+on: [pull_request]
+
+jobs:
+  gradle-check:
+    runs-on: ubuntu-latest
+    timeout-minutes: 130
```
Comment (owaiskazi19) [inline .github/workflows/gradle-check.yml:7]: As the last PR was closed. 
How are we deciding the timeout here to be 130? Did we document it somewhere?

Comment (peterzhuamazon) [reply .github/workflows/gradle-check.yml:7]: Offline discussion with @owaiskazi19 @dreamer-89 in this url: https://github.com/opensearch-project/opensearch-build/issues/851#issuecomment-1167932678

**Comment by peterzhuamazon** [comment]
Added here in case it failed again:

Gradle check on main branch success after changing to 128CPU setup:
https://github.com/opensearch-project/OpenSearch/runs/7081855652?check_suite_focus=true
https://build.ci.opensearch.org/job/gradle-check/14/

**Review by dreamer-89 — APPROVED** [review]
LGTM!

**Comment by owaiskazi19** [comment]
> ❌   Gradle Check failure [5f599f6](https://github.com/opensearch-project/OpenSearch/commit/5f599f6a284ae21b27674ffc34f661e8b41abb68) [Log 6384](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6384.log)
> 
> [Reports 6384](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6384_reports.zip)

@peterzhuamazon are we planning to remove ci-bot comments as a part of this PR?

**Comment by peterzhuamazon** [comment]
> > ❌   Gradle Check failure [5f599f6](https://github.com/opensearch-project/OpenSearch/commit/5f599f6a284ae21b27674ffc34f661e8b41abb68) [Log 6384](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6384.log)
> > [Reports 6384](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6384_reports.zip)
> 
> @peterzhuamazon are we planning to remove ci-bot comments as a part of this PR?

This is old workflow @owaiskazi19 have nothing to do with the new one.

**Comment by owaiskazi19** [comment]
> > > ❌   Gradle Check failure [5f599f6](https://github.com/opensearch-project/OpenSearch/commit/5f599f6a284ae21b27674ffc34f661e8b41abb68) [Log 6384](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6384.log)
> > > [Reports 6384](https://ci.opensearch.org/logs/ci/workflow/OpenSearch_CI/PR_Checks/Gradle_Check/gradle_check_6384_reports.zip)
> > 
> > 
> > @peterzhuamazon are we planning to remove ci-bot comments as a part of this PR?
> 
> This is old workflow @owaiskazi19 have nothing to do with the new one.

Yes. So are we planning to remove the old workflow or do we have any issue to track it?

**Comment by peterzhuamazon** [comment]
@dreamer-89 new workflow has much higher stable rate compares to old one, success again.
https://build.ci.opensearch.org/job/gradle-check/15/console

The old one is failure miserably during a test, flaky integTest.

**Comment by peterzhuamazon** [comment]
Seems like still need to present on these old branches, backport them.

**Comment by dblock** [comment]
Could we try https://github.com/marketplace/actions/trigger-jenkins-job instead of huge amount of boilerplate scripting here?

**Pull request merged**